### ISSUE TITLE
Can't get nickname to show in animated tablist plugins' tablist 

### ISSUE THREAD

**Issue by axemwa** [spyproof/NicknameManager#1]
Is there a work around for this or something I can fix?

**Comment by axemwa** [comment]
I suggest maybe making the plugin replace the display name instead like however essentials does it or is there another way?

**Comment by spyproof** [comment]
I am assuming this is about Bukkit and not Sponge. It really depends on the animated tabs plugin. It should extract the nickname from Player.getDisplayName() in stead of getting it directly from.

As you can see in [my code](https://github.com/spyproof/NicknameManager/blob/master/Bukkit/src/main/java/be/spyproof/nickmanager/listeners/PlayerListener.java), I am setting the display name. So there is no direct dependancy needed with my plugin.

```java
@EventHandler
public void onLogin(PlayerJoinEvent event)
{
    NicknameData nicknameData = this.playerController.wrap(event.getPlayer().getUniqueId(), event.getPlayer().getName());
    if (nicknameData.getNickname().isPresent())
        event.getPlayer().setDisplayName(ChatColor.translateAlternateColorCodes('&', nicknameData.getNickname().get()) + ChatColor.RESET);
}
```

**Comment by montlikadani** [comment]
@spyproof 
You must use the `p.setPlayerListName(name)` method to change the player tab name.

**Comment by spyproof** [comment]
The problem with that method is that there is a 16 character limit, including color codes. So if you want your username to be red, you have to cut off the last 2 characters if the username is 16 characters long.

Because of this behaviour it never got implemented and i believe it to be correct.

Do note that this plugin is free and opensource, you can fork my code and make any adjustments you like to it.

**Comment by montlikadani** [comment]
No, there is no character limit for `p.setPlayerListName()`. The character limit is set to `setPrefix()`, `setSuffix()`, and `setDisplayName()`. You can check it here: https://hub.spigotmc.org/stash/projects/SPIGOT/repos/bukkit/browse/src/main/java/org/bukkit/scoreboard/Team.java

I know and saw the plugin open source, but I'm not gonna touch it yet.

What is currently a issue here does not matter, as this can be disabled in my plugin for one of the features.

**Comment by spyproof** [comment]
Interesting, this was developed back on MC 1.7.10, the behaviour must have changed by now, there doesnt seem to be a character limit, tested up to 256 and no problems.

I don't mind adding this as a new feature that can be enabled/disabled with a config option.

**Comment by spyproof** [comment]
Check out the new version: https://github.com/spyproof/NicknameManager/releases/tag/1.1.0

**Comment by axemwa** [comment]
it works in 1.12 but I still have to remove the space plugin name in the plugin.yml of the jar file in order to get it to load, it doesn't seem to like the spaces in the plugin name for some reason

**Comment by axemwa** [comment]
also when I log out and back in, it removes the new nickname from the playerlist and reverts back to the essentials nickname

**Comment by axemwa** [comment]
Also this error shows up in console now when attempting to set a blacklisted name instead of just saying insufficent permission ingame etc

`[11:24:06 INFO]: Ayef_Keing issued server command: /nick set admin [11:24:06 INFO]: [ChatControl 7.9.17] Command event took 179 ms [11:24:06 ERROR]: Unhandled exception executing command 'nick' in plugin NicknameManager v1.0.0 org.bukkit.command.CommandException: Unhandled exception executing command 'nick' in plugin NicknameManager v1.0.0 at org.bukkit.command.PluginCommand.execute(PluginCommand.java:46) ~[patched_1.12.2.jar:git-Paper-1554] at com.earth2me.essentials.Essentials.onCommandEssentials(Essentials.java:446) ~[?:?] at com.earth2me.essentials.Essentials.onCommand(Essentials.java:435) ~[?:?] at org.bukkit.command.PluginCommand.execute(PluginCommand.java:44) ~[patched_1.12.2.jar:git-Paper-1554] at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:151) ~[patched_1.12.2.jar:git-Paper-1554] at gyurix.commands.CustomCommandMap.dispatch(CustomCommandMap.java:82) ~[?:?] at org.bukkit.craftbukkit.v1_12_R1.CraftServer.dispatchCommand(CraftServer.java:685) ~[patched_1.12.2.jar:git-Paper-1554] at net.minecraft.server.v1_12_R1.PlayerConnection.handleCommand(PlayerConnection.java:1476) ~[patched_1.12.2.jar:git-Paper-1554] at net.minecraft.server.v1_12_R1.PlayerConnection.chat(PlayerConnection.java:1385) ~[patched_1.12.2.jar:git-Paper-1554] at net.aufdemrand.denizen.nms.impl.packets.handlers.AbstractListenerPlayIn_v1_12_R1.chat(AbstractListenerPlayIn_v1_12_R1.java:71) ~[?:?] at org.bukkit.craftbukkit.v1_12_R1.entity.CraftPlayer.chat(CraftPlayer.java:352) ~[patched_1.12.2.jar:git-Paper-1554] at com.cyr1en.cp.listener.Prompt$1.run(Prompt.java:85) ~[?:?] at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftTask.run(CraftTask.java:63) ~[patched_1.12.2.jar:git-Paper-1554] at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:423) ~[patched_1.12.2.jar:git-Paper-1554] at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:840) ~[patched_1.12.2.jar:git-Paper-1554] at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:423) ~[patched_1.12.2.jar:git-Paper-1554] at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:774) ~[patched_1.12.2.jar:git-Paper-1554] at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:666) ~[patched_1.12.2.jar:git-Paper-1554] at java.lang.Thread.run(Thread.java:748) [?:1.8.0_172] Caused by: java.lang.NullPointerException at be.spyproof.nickmanager.util.BukkitUtils.acceptedRules(BukkitUtils.java:53) ~[?:?] at be.spyproof.nickmanager.commands.player.AbstractPlayerCmd.execute(AbstractPlayerCmd.java:31) ~[?:?] at be.spyproof.nickmanager.commands.ParentCmd.execute(ParentCmd.java:93) ~[?:?] at be.spyproof.nickmanager.commands.ICommand.onCommand(ICommand.java:23) ~[?:?] at org.bukkit.command.PluginCommand.execute(PluginCommand.java:44) ~[patched_1.12.2.jar:git-Paper-1554] ... 18 more`

**Comment by axemwa** [comment]
the config seemed to keep on getting corrupted when making changes to the config then restarting that's wat was causing the error I think

**Comment by axemwa** [comment]
after stopping adjusting config then starting instead seems to be succesful and working fine

**Comment by axemwa** [comment]
sorry I was trying to save time by putting it all into the same issue I have made seperate the new issues

**Comment by axemwa** [comment]
Hey guys I noticed that  there are instances where the NicknameManager's nickname won't appear in animated Tablist when a player logs in but when leaving and rejoining the server the nickname applies as it should, it this animated tablist causing this or nickname manager?
